raddb/mods-available/yubikey

```diff
@@ -22,33 +22,84 @@ yubikey {
 	#
 	#  Decrypt mode - Tokens will be decrypted and processed locally
 	#
+	#  In the authorize section, this module should be placed before the
+	#  persistent storage module.
+	#
+	#  If the account's use of the OTP is required, then the
+	#  yubikey_authorize_otpkey policy should be placed directly after the
+	#  authorize section's persistent storage module.
+	#
+	#  The authenticate section should have the following added:
+	#	 Auth-Type yubikey {
+	#	#	pap
+	#		yubikey
+	#	#	yubikey_authenticate_id
+	#	#	yubikey_authenticate_counter
+	#	 }
+	#
+	#  If two factor authentication (password concatenated with OTP) is
+	#  required, split configuration should be set to yes and the pap module
+	#  should be uncommented. Each account will need a password added (either
+	#  using attribute Cleartext-Password, SSHA-Password, etc.).
+	#
+	#  If the suggested check of the Yubikey-Private-ID for validation is
+	#  required, uncomment the yubikey_authenticate_id policy. Each account
+	#  will need the Yubikey-Private-ID attribute added.
+	#
+	#  If the suggested check of the Yubikey-Counter for replay attack is
+	#  required, uncomment the yubikey_authenticate_counter policy. Each
+	#  account will need the Yubikey-Counter attribute added.
+	#
 	#  The module itself does not provide persistent storage as this
 	#  would be duplicative of functionality already in the server.
+	#  See the post-auth section below.
 	#
-	#  Yubikey authentication needs two control attributes
+	#  Yubikey authentication requires the following control attribute
 	#  retrieved from persistent storage:
 	#    * Yubikey-Key     - The AES key used to decrypt the OTP data.
-	#                        The Yubikey-Public-Id and/or User-Name
-	#                        attributes may be used to retrieve the key.
-	#    * Yubikey-Counter - This is compared with the counter in the OTP
-	#                        data and used to prevent replay attacks.
-	#                        This attribute will also be available in
-	#                        the request list after successful
-	#                        decryption.
-	#
-	#  Yubikey-Counter isn't strictly required, but the server will
-	#  generate warnings if it's not present when yubikey.authenticate
-	#  is called.
+	#                        The value is a 16-byte binary blob.
 	#
 	#  These attributes are available after authorization:
 	#    * Yubikey-Public-ID  - The public portion of the OTP string
+	#                           The value is a id_len modhex string.
+	#
+	#  The Yubikey-Public-Id and/or User-Name attributes may be used to
+	#  retrieve  the Yubikey-Key, Yubikey-Private-ID, etc. from persistant
+	#  st rage. For the sql module, modify the main queries from:
```

Comment:
Is that a metal band (st rage)? What do I have to do to retrieve the keys from them? I'm not down with biting the heads off bats.